horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/RunServiceTest.java

```diff
@@ -59,31 +68,219 @@
+      Test test = createTest(exampleTest);
+
+      dataSetQueue.drainTo(new ArrayList<>());
+      uploadRun(runWithValue(42.0d, acmeSchema), test.name);
+
+      DataSet event = dataSetQueue.poll(10, TimeUnit.SECONDS);
+//      DataSet event = dataSetQueue.poll(1800, TimeUnit.SECONDS);
+
+      assertNotNull(event);
+      JsonNode node = event.data;
+      assertEquals(1, node.size());
+      JsonNode schema = node.path("$schema");
+      assertEquals("urn:AcmeInc:AcmeRrUs:1.0", schema.textValue());
+      JsonNode value = node.path("value");
+      assertEquals("value", value.textValue());
    }
 
+//   @org.junit.jupiter.api.Test
+   public void testTransformationUsingSameSchemaInBothLevelsTestWithoutTransformer(TestInfo info) throws InterruptedException {
+      // expecting 1 dataset for each schema
+      BlockingQueue<DataSet> dataSetQueue = eventConsumerQueue(DataSet.class, DataSet.EVENT_NEW);
+
+      Schema acmeSchema = createExampleSchema("AcmeCorp", "AcmeInc", "AcmeRrUs", false);
+
+      Test exampleTest = createExampleTest(getTestName(info));
+      Test test = createTest(exampleTest);
+
+      dataSetQueue.drainTo(new ArrayList<>());
+      uploadRun(runWithValue(42.0d, acmeSchema, acmeSchema), test.name);
+
+    DataSet event = dataSetQueue.poll(10, TimeUnit.SECONDS);
+//      DataSet event = dataSetQueue.poll(1800, TimeUnit.SECONDS);
+
+      assertNotNull(event);
+      JsonNode node = event.data;
+      assertTrue(node.isArray());
+      assertEquals(2 , node.size());
+      JsonNode first = node.path(0).path("$schema");
+      assertEquals("urn:AcmeInc:AcmeRrUs:1.0", first.textValue());
+      JsonNode second = node.path(0).path("$schema");
+      assertEquals("urn:AcmeInc:AcmeRrUs:1.0", second.textValue());
+   }
+
+   public void testTransformationUsingSingleSchemaTransformersProcessScalarPlusArray(TestInfo info) throws InterruptedException {
+      BlockingQueue<DataSet> dataSetQueue = eventConsumerQueue(DataSet.class, DataSet.EVENT_NEW);
+
+      Schema schema = createExampleSchema("ArrayCorp", "ArrayInc", "ArrayRrUs", false);
+      NamedJsonPath arrayPath = createExampleNamedJsonPath("mheep", "$.values");
+      String arrayFunction = createArrayFunction("mheep");
+
+      NamedJsonPath scalarPath = createExampleNamedJsonPath("sheep", "$.value");
+      String scalarFunction = createFunction("sheep");
+
+      Transformer arrayTransformer = createExampleTransformerWithJsonPath("arrayT", schema, arrayFunction, arrayPath);
+      Transformer scalarTransformer = createExampleTransformerWithJsonPath("scalarT", schema, scalarFunction, scalarPath);
+
+      Test exampleTest = createExampleTest(getTestName(info));
+      Test test = createTest(exampleTest);
+      addTransformer(test, arrayTransformer, scalarTransformer);
+
+      ObjectNode data = runWithValue(42.0d, schema);
+      ArrayNode array = JsonNodeFactory.instance.arrayNode();
+      array.add(42).add(43).add(44);
+      data.set("values", array);
```

Comment:
looks like you're adding these values outside of any schema

Reply:
yes got distracted. WIP